conda/signature.py

```diff
@@ -0,0 +1,66 @@
+import sys
+import base64
+import hashlib
+from os.path import abspath, expanduser, join
+
+from conda.compat import PY3
+
+from Crypto.PublicKey import RSA
+
+
+KEYS = {}
+KEYS_DIR = abspath(expanduser('~/.conda/keys'))
+
+
+def sig2ascii(i):
+    ret = []
+    while i:
+        i, r = divmod(i, 256)
+        ret.append(r)
+    if PY3:
+        s = bytes(n for n in ret[::-1])
+    else:
+        s = ''.join(chr(n) for n in ret[::-1])
+    return base64.b64encode(s).decode('utf-8')
+
+
+def ascii2sig(s):
+    res = 0
+    for c in base64.b64decode(s):
+        res *= 256
+        res += (c if PY3 else ord(c))
+    return res
+
+
+def hash_file(path):
+    h = hashlib.new('sha256')
+    with open(path, 'rb') as fi:
+        while True:
+            chunk = fi.read(262144)
+            if not chunk:
+                break
+            h.update(chunk)
+    return h.digest()
+
+
+def verify(path):
+    """
+    Verify the file `path`, with signature `path`.sig, against the key
+    found under ~/.conda/keys/<key_name>.pub
+    """
+    try:
+        with open(path + '.sig') as fi:
+            key_name, sig = fi.read().split()
+    except IOError:
+        return 'NO_SIGATURE'
```

Comment:
Is this function intended to only return one of three values "NO_SIGNATURE", "VALID", "INVALID"?  It seems to me that it's actually not very helpful to swallow all types of IOErrors and convert them into a single, opaque "NO_SIGNATURE" string.

If the rationale for swallowing exceptions is to make this function a simple utility that can be relied upon to return only one of three values, then this should also catch potential `ValueError`s resulting from the tuple unpacking of the call to `split()`.